### PR TITLE
Fix NavigationCamera infinite recursion on animation cancel and update device farm config

### PR DESCRIPTION
**Pull request by zeac**
### Description
Switching between camera states during animation can cause a StackOverflow crash. Here is the test and the fix for it.

### Changelog
```
<changelog>Fixed stack overflow error that could happen when `NavigationCamera` state transitions were canceled while running on some Android API levels (definitely 21-24 and maybe more).</changelog>
```

Fixes https://github.com/mapbox/mapbox-navigation-android/issues/4453.

### CODE REVIEW

**Review by VitaminPSG — APPROVED** [review]

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/SanityUiRouteTest.kt [hunk 1]
```diff
@@ -16,4 +17,29 @@ class SanityUiRouteTest : SimpleMapViewNavigationTest() {
         Espresso.onIdle()
         arrivalIdlingResource.unregister()
     }
+
+    @Test
+    fun navigation_camera_mode_changes_completes() {
```
Comment (LukasPaczos) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/SanityUiRouteTest.kt:22]: I reverted the changes introduced in this PR and the provided test does not reproduce a StackOverflow on a Pixel 2 running Android 11 for me. Looking more into it.

Comment (LukasPaczos) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/SanityUiRouteTest.kt:22]: Is this test crashing on your device without the other changes from the PR? What device and Android version are you running on?

**Comment by LukasPaczos** [comment]
Is this PR targetting https://github.com/mapbox/mapbox-maps-android/issues/409 or https://github.com/mapbox/mapbox-navigation-android/issues/4453? The above comment mentions https://github.com/mapbox/mapbox-maps-android/issues/409 but I don't see how that's related.

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/SanityUiRouteTest.kt (hunk 1 above)
Comment (zeac) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/SanityUiRouteTest.kt:22]: It seems this recursion has also been fixed on Android side here: https://cs.android.com/android/_/android/platform/frameworks/base/+/3dbaae1ef4f221b3626810f4ba19eec068dd6304
I'm not sure in what Android it was fixed, 6.0.1 has this bug.

**Comment by zeac** [comment]
> Is this PR targetting [mapbox/mapbox-maps-android#409](https://github.com/mapbox/mapbox-maps-android/issues/409) or #4453? The above comment mentions [mapbox/mapbox-maps-android#409](https://github.com/mapbox/mapbox-maps-android/issues/409) but I don't see how that's related.

They both about the single problem. Stacktrace in [mapbox/mapbox-maps-android#409](https://github.com/mapbox/mapbox-maps-android/issues/409) is for direct outcome of stackoverflow in our app.

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/SanityUiRouteTest.kt (hunk 1 above)
Comment (LukasPaczos) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/SanityUiRouteTest.kt:22]: Looks like this is it, I was able to reproduce on Android 21 :+1: I'll tweak the PR to make sure that the test targets a specific API level and that we have a device on the farm that can catch this.

**Review by LukasPaczos — APPROVED** [review]
:ship: thanks @zeac!

**Comment by LukasPaczos** [comment]
I also pushed a commit here that updates the devices that we use on the Firebase device farm since some of them were deprecated and some were already removed.

**Pull request merged**